annotation/annotation/annotations/resources.py

```diff
@@ -288,6 +278,16 @@ def post_annotation_by_pipeline(
         .order_by(desc(AnnotatedDoc.date))
         .first()
     )
+    # Add categories to jobs microservice for any commit to keep
+    # it consistent (https://github.com/epam/badgerdoc/issues/841)
+    categories = combine_categories(doc.categories, doc.pages)
+    category_objects = fetch_bunch_categories_db(
+        db, categories, x_current_tenant, root_parents=True
+    )
+    update_jobs_categories(db, job_id, category_objects)
+    jobs_communication.append_job_categories(
+        job_id, list(categories), x_current_tenant, token.token
```

Comment:
`combine_categories` already returns list of categories, I don't think we need explicit type cast here
```suggestion
        job_id, categories, x_current_tenant, token.token
```

Reply:
`combine_categories` function returns a `Set[str]`. However, we need to serialize it for passing as a JSON string, so here we're explicitly converting from set to list. It seems that your suggestion might break the downstream code.